### PR TITLE
RET 3090 - Bundles- About your hearing documents

### PR DESCRIPTION
**Pull request by brendanmchugh**
### JIRA link
https://tools.hmcts.net/jira/browse/RET-3090

### Change description

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/definitions/api/caseApiResponse.ts [hunk 1]
```diff
@@ -65,6 +65,61 @@ export interface CaseData {
...
+    Hearing_notes?: string;
+    Hearing_date_start?: Date;
+    Hearing_judge_name?: string;
+    Hearing_venue?: {
+      value: HearingVenueItem;
+      list_items: HearingVenueItem[];
+      selectedCode: string;
+      selectedLabel: string;
+    };
+    hearingDateCollection: {
```
Comment (ikirsanov) [inline src/main/definitions/api/caseApiResponse.ts:96]: this collection is mandatory? or it could be with question mark?

Comment (brendanmchugh) [reply src/main/definitions/api/caseApiResponse.ts:96]: I think the hearingDateCollection is mandatory. A hearing needs to have at least one date.

File: src/main/controllers/AboutHearingDocumentsController.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  private aboutHearingDocumentsContent: FormContent;
+
+  constructor() {
+    // intentionally empty
+  }
+
+  public post = async (req: AppRequest, res: Response): Promise<void> => {
+    const { userCase } = req.session;
+    req.session.errors = [];
+    const errorsForPage = () => {
```
Comment (marcjudge) [inline src/main/controllers/AboutHearingDocumentsController.ts:27]: Please move this out of the controller into a helper class if possible

File: src/main/controllers/AboutHearingDocumentsController.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        day: 'numeric',
+        month: 'short',
+        year: 'numeric',
+      }).format(new Date(rawDate));
+
+    if (!req.session?.userCase?.hearingCollection?.length) {
+      logger.info('no hearing collection found, redirecting to citizen hub');
+      return res.redirect(`/citizen-hub/${req.session.userCase.id}${getLanguageParam(req.url)}`);
+    }
+    const radioBtns = req.session?.userCase?.hearingCollection
```
Comment (marcjudge) [inline src/main/controllers/AboutHearingDocumentsController.ts:64]: Same with this, please move to a helper class

**Review by ikirsanov — DISMISSED** [review]

File: src/main/controllers/AboutHearingDocumentsController.ts (hunk 2 above)
Comment (brendanmchugh) [reply src/main/controllers/AboutHearingDocumentsController.ts]: this has been moved.

File: src/main/controllers/AboutHearingDocumentsController.ts (hunk 3 above)
Comment (brendanmchugh) [reply src/main/controllers/AboutHearingDocumentsController.ts]: this has been moved.

**Review by marcjudge — APPROVED** [review]

**Pull request merged**